jqjq.jq

```diff
@@ -2456,35 +2478,85 @@ def fromjqtest:
+      elif $a == "-M" or $a == "--monochrome-output" then {monochrome_output: true}, (.[1:] | _f)
+      elif $a == "--run-tests"                       then {run_tests: true}, (.[1:] | _f)
+      elif $a == "--"                                then {filter: .[1]}, (.[2:] | _f)
+      elif $a | startswith("-")                      then error("unknown argument: \($a)")
+      else {filter: $a}, (.[1:] | _f)
       end;
     ( [_f]
     | add
     );
 
+  # get the ANSI color codes for printing values
+  # corresponds to jv_set_colors in jq and its usage in main
+  def _parse_colors($opts; $env):
+    # color order: null, false, true, number, string, array, object, field
+    ( ["0;90", "0;39", "0;39", "0;39", "0;32", "1;39", "1;39", "1;34"] as $default
+    | if $env | has("JQ_COLORS") then
+        # only up to the first 8 color sequences are used
+        ( ($env.JQ_COLORS | split(":")[:8]) as $custom
+        # jq limits color sequences to 12 bytes, to fit into 16-byte buffers
+        # with ESC, [, m, and NUL.
+        | if $custom | all(length <= 12 and test("^[0-9;]*$")) then
+            $custom + $default[$custom | length:]
+          else
+            "Failed to set $JQ_COLORS\n" | stderr | $default
+          end
+        )
+      else $default
+      end
+    | if $opts.monochrome_output or
+          (($opts.color_output | not) and ($env.NO_COLOR | . != null and . != "")) then
+        null
+      end
+    );
+
+  def _parse_opts($opts; $env):
+    ( $opts
+      + { colors: _parse_colors($opts; $env)
+        , indent: (if $opts.compact_output then 0 else 2 end)
+        }
+    );
+
   def _help:
     ( "jqjq - jq implementation of jq"
     , "Usage: jqjq [OPTIONS] [--] [EXPR]"
-    , "  --jq PATH        jq implementation to run with"
-    , "  --lex            Lex EXPR"
-    , "  --no-builtins    Don't include builtins"
-    , "  --null-input,-n  Null input"
-    , "  --parse          Lex then parse EXPR"
-    , "  --repl           REPL"
-    , "  --run-tests      Run jq tests from stdin"
-    , "  --slurp,-s       Slurp inputs into an array"
+    , ""
+    , "Options:"
+    , "  --jq PATH                 jq implementation to run with"
+    , "  --lex                     Lex EXPR"
+    , "  --no-builtins             Don't include builtins"
+    , "  --parse                   Lex then parse EXPR"
+    , "  --repl                    REPL"
+    , ""
+    , "  --null-input / -n         Null input"
+    , "  --compact-output / -c     Output each object on one line"
+    , "  --raw-output / -r         Output strings raw with newline"
+    , "  --raw-output0             Output strings raw with NUL"
+    , "  --join-output             Output strings raw"
+    , "  --slurp / -s              Slurp inputs into an array"
+    , "  --color-output / -C       Force colored output"
+    , "  --monochrome-output / -M  Disable colored output"
+    , "  --run-tests               Run jq tests from stdin"
```

Comment:
Sort them?

Reply:
I went by the order in the jq manual (which matches `man jq`), because that's how I think of them. I double-checked and I put `--slurp` in the wrong order for that (it should be be between `--null-input` and `--compact-output`), so I could fix that one. Or I could alphabetize it.

Reply:
Ok, no strong opinion really, i've started to just sort thing alphabetically to not have to ponder about it 😄

Reply:
Well if that weak opinion becomes stronger later, feel free to alphabetize it 😅. I reordered --slurp.